src/ui/Field.hpp

```diff
@@ -0,0 +1,98 @@
+//! @file Field.hpp
```

Comment:
Maybe something cool we could do sometime is add decorators for these...actually, maybe all widgets could do with that.